TSPL.docc/LanguageGuide/Concurrency.md

```diff
@@ -1591,6 +1591,370 @@ You can also use an unavailable conformance
+from a concurrent task results in an error,
+because it would allow calling the main actor isolated implementation
+of `Dancer.perform` from outside the main actor.
+
+Abstract code that uses type parameters and `any` types
+can check whether a value conforms to a protocol
+through dynamic casting.
+The following code has a protocol `Dancer`,
+and a method `performIfDancer` that accepts a parameter of type `Any`
+which is dynamically cast to `any Dancer` in the function body:
+
+```swift
+protocol Dancer {
+    func perform()
+}
+
+func performIfDancer(_ value: Any) {
+    if let dancer = value as? any Dancer {
+        dancer.perform()
+    }
+}
+```
+
+Isolated conformances are only safe to use
+when the code is running on the global actor
+that the conformance is isolated to,
+so the dynamic cast only succeeds
+if the dynamic cast occurs on that global actor.
+For example, if you declare a main-actor isolated conformance to `Dancer`
+and call `performIfDancer` with an instance of the conforming type,
+the dynamic cast will succeed
+when `performIfDancer` is called in a main actor task, and it
+will fail when `performIfDancer` is called in a concurrent
+task:
+
+```swift
+@MainActor class Ballerina: @MainActor Dancer {
+    func perform() {
+        print("Ballerina.perform")
+    }
+}
+
+Task { @MainActor in
+    let ballerina = Ballerina()
+    performIfDancer(ballerina)  // Prints "Ballerina.perform"
+}
+
+Task { @concurrent in
+    let ballerina = Ballerina()
+    performIfDancer(ballerina)  // Prints nothing
+}
+```
+
+In the above code,
+the call to `performIfDancer` from a main-actor isolated task
+matches the isolation of the conformance,
+so the dynamic cast succeeds.
+The call to `performIfDancer` from a concurrent task
+happens outside the main actor,
+so the dynamic cast fails and `perform` is not called.
+
+#### Restricting Isolated Conformances in Concurrent Code
+
+Protocol requirements can be used
+through instances of conforming types and through
+instances of the conforming types themselves,
+called *metatype values*.
+In generic code,
+a conformance requirement to `Sendable` or `SendableMetatype`
+tells Swift that an instance or metatype value is safe to use concurrently.
+To prevent isolated conformances from being used outside of their actor,
+a type with an isolated conformance
+can't be used as the concrete generic argument for a type
+parameter that requires a conformance
+to `Sendable` or `SendableMetatype`.
+
+A conformance requirement to `Sendable` indicates
+that instances may be passed across isolation boundaries and used concurrently:
+
+```swift
+protocol Dancer {
+    func perform()
+}
+
+func performConcurrently<D: Dancer>(_ dancer: D) where D: Sendable {
+    Task { @concurrent in
+        dancer.perform()
+    }
+}
+```
+
+The above code would admit data races
+if the conformance to `Dancer` was isolated,
+because the implementation of `perform`
+may access global actor isolated state.
```

Comment:
I'm having a hard time with this sentence, probably because of the irrealis mood on the verbs (would admit, was isolated, may access).  Are you saying that the code above is safe, and then describing why by telling me about something that can't happen?  Or are you describing a scenario where the code has possible data races?